addons/account_update_tax_tags/tests/test_account_update_tax_tags_wizard.py

```diff
@@ -0,0 +1,325 @@
+                        'test_tag_refund_tax'
+                    ])
+                    move = self._create_move(move_type=move_type, tax_ids=tax_2)
+                    super_type = move_type.split('_')[1]
+                    if super_type == 'receipt':
+                        super_type = 'invoice'  # receipt type acts just like invoice one
+                    self._change_tax_tag(tax_2, f'{move_type}_{line_type}_tag_changed', invoice=super_type == 'invoice', base=line_type == 'base')
+                    self.wizard.update_amls_tax_tags()
+
+                    invoice_line, tax_line, counterpart_line = self._get_amls_by_type(move)
+                    if line_type == 'base':
+                        self.assertEqual(invoice_line.tax_tag_ids.name, f'{move_type}_{line_type}_tag_changed')  # should change
+                        self.assertEqual(tax_line.tax_tag_ids.name, f'test_tag_{super_type}_tax')  # should not change
+                        self.assertFalse(counterpart_line.tax_tag_ids)  # should not change
+                    else:
+                        self.assertEqual(invoice_line.tax_tag_ids.name, f'test_tag_{super_type}_base')  # should not change
+                        self.assertEqual(tax_line.tax_tag_ids.name, f'{move_type}_{line_type}_tag_changed')  # should change
+                        self.assertFalse(counterpart_line.tax_tag_ids)  # should not change
+
+    def test_update_move_type_entry(self):
+        """ Test that move of type 'entry' are correctly updated.
+        If a move has negative balance and use a sale tax, it should act as an invoice.
+        If a move has positive balance and use a sale tax, it should act as a refund.
+        If a move has negative balance and use a purchase tax, it should act as an invoice.
+        If a move has positive balance and use a purchase tax, it should be treated as an invoice.
+        """
+        account = self.company_data['default_account_assets']  # Account is not relevant for the test but must be set.
+        for tax_type in ['sale', 'purchase']:
+            tax = self._create_tax(f'test_{tax_type}_tax', amount=10, type_tax_use=tax_type, tag_names=self.tag_names)
+            for balance in [-1000, 1000]:
+                with self.subTest(f'Testing move type entry {tax_type}-{balance}'):
+
+                    # Create one entry and it's reverse.
+                    move_form = Form(self.env['account.move'].with_context(default_move_type='entry'))
+                    with move_form.line_ids.new() as line:
+                        line.account_id = account
+                        if balance < 0:
+                            line.credit = abs(balance)
+                        else:
+                            line.debit = abs(balance)
+                        line.tax_ids.add(tax)
+
+                    # Create a third account.move.line for balance. It will be automatically filled given above lines.
+                    move_form.line_ids.new().save()
```

Comment:
Doesn't it do it automatically? :o

Reply:
Do we want to rely on a form here anyway ? Seems a mere create() would do the trick